tutorials/0_to_litgpt.md

```diff
@@ -0,0 +1,549 @@
+
+```
+git clone https://github.com/Lightning-AI/litgpt.git
+cd litgpt
+pip install -e '.[all]'
+```
+
+
+&nbsp;
+## Pretrain LLMs
+
+Finetuning LLMs requires substantial compute resources and time commitment. For that reason, most researchers and practitioners prefer to skip this step and continue with the Download pretrained model weights section instead. 
+
+However, if you feel adventurous and want to pretrain your own LLM, here's how.
+
+First, we have to decide which type of model architecture we want to use. We list the available architectures by using the `pretrain` command without any additional arguments:
+
+```bash
+litgpt pretrain
+```
+
+This prints a list of all available model architectures in alphabetical order:
+
+```
+Camel-Platypus2-13B
+Camel-Platypus2-70B
+CodeLlama-13b-Python-hf
+...
+tiny-llama-1.1b
+vicuna-13b-v1.3
+vicuna-13b-v1.5
+vicuna-13b-v1.5-16k
+vicuna-33b-v1.3
+vicuna-7b-v1.3
+vicuna-7b-v1.5
+vicuna-7b-v1.5-16k
+```
+
+Suppose we want to pretraining the 1.1B parameter small `tiny-llama-1.1b` model. Before starting finetuning, we must also choose and download a tokenizer. 
+
+We can download a tokenizer via the `download` command. Note that running `litgpt download` without any additional arguments will also print a list of all available models and tokenizers to download. 
+
+To filter for specific models, e.g., TinyLlama, we can use the `grep` command in our terminal:
+
+```bash
+litgpt download | grep  TinyLlama
+```
+
+This prints
+
+```
+TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+TinyLlama/TinyLlama-1.1B-Chat-v1.0
+```
+
+Let's now download the tokenizer corresponding to `TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T` that we can then use to pretrain the TinyLlama model, which saves the download tokenizer to a `checkpoints/` folder by default:
+
+```
+litgpt download \
+   --repo_id TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T \
+   --tokenizer_only true
+```
+
+&nbsp;
+
+<img src="images/0_to_litgpt/pretrain.webp" width=400>
+
+&nbsp;
+
+Next, we can pretrain the model on the OpenWebText dataset with the default setting as follows:
+
+```bash
+litgpt pretrain \
+  --model_name tiny-llama-1.1b \
+  --data OpenWebText \
+  --tokenizer_dir checkpoints/TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+```
+
+If you are interested in additional settings, you can use the help command as follows:
+
+```
+litgpt pretrain --help
+```
+
+&nbsp;
+
+> [!TIP]
+> Above, we only covered the most basic commands for pretraining a model using LitGPT. We highly recommend checking the resources below if you are interested in pretraining a model.
+
+&nbsp;
+
+**More information and additional resources**
+
+- [tutorials/pretrain_tinyllama](./pretrain_tinyllama.md): A tutorial for finetuning a 1.1B TinyLlama model on 3 trillion tokens
+- [config_hub/pretrain](../[config_hub/pretrain): Pre-made config files for pretraining that work well out of the box
```

Comment:
```suggestion
- [config_hub/pretrain](../config_hub/pretrain): Pre-made config files for pretraining that work well out of the box
```

Reply:
Oof. I actually got glasses ... now I just need to wear them :D